server/armeria-server/cats/src/main/scala/sttp/tapir/server/armeria/cats/TapirCatsService.scala

```diff
@@ -0,0 +1,98 @@
+package sttp.tapir.server.armeria.cats
+
+import cats.effect.Async
+import cats.effect.std.Dispatcher
+import com.linecorp.armeria.common.{HttpData, HttpRequest, HttpResponse}
+import com.linecorp.armeria.server.ServiceRequestContext
+import fs2.{Chunk, Stream}
+import fs2.interop.reactivestreams._
+import java.util.concurrent.CompletableFuture
+import org.reactivestreams.Publisher
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+import sttp.capabilities.fs2.Fs2Streams
+import sttp.monad.MonadError
+import sttp.monad.syntax._
+import sttp.tapir.integ.cats.CatsMonadError
+import sttp.tapir.server.ServerEndpoint
+import sttp.tapir.server.armeria._
+import sttp.tapir.server.interpreter.{BodyListener, ServerInterpreter}
+
+private[cats] final case class TapirCatsService[F[_]: Async](
+    serverEndpoints: List[ServerEndpoint[Fs2Streams[F], F]],
+    armeriaServerOptions: ArmeriaCatsServerOptions[F]
+) extends TapirService[Fs2Streams[F], F] {
+
+  private[this] val dispatcher: Dispatcher[F] = armeriaServerOptions.dispatcher
+  private[this] val fs2StreamCompatible: StreamCompatible[Fs2Streams[F]] = Fs2StreamCompatible(dispatcher)
+
+  private[this] implicit val monad: MonadError[F] = new CatsMonadError()
+  private[this] implicit val bodyListener: BodyListener[F, ArmeriaResponseType] = new ArmeriaBodyListener
+
+  override def serve(ctx: ServiceRequestContext, req: HttpRequest): HttpResponse = {
+    implicit val ec: ExecutionContext = ExecutionContext.fromExecutorService(ctx.eventLoop())
+
+    val catsFFromFuture = new CatsFromFuture
+    val serverRequest = new ArmeriaServerRequest(ctx)
+    val interpreter = new ServerInterpreter(
```

Comment:
probably a minor optimization, but would it be possible to create the `ServerInterpreter` once when the `TapirCatsService` is created, not for every request?